persistence-cassandra/javadsl/src/main/scala/com/lightbend/lagom/javadsl/persistence/cassandra/CassandraSession.scala

```diff
@@ -53,56 +46,12 @@ final class CassandraSession(system: ActorSystem, settings: CassandraSessionSett
       ))
     )
 
-  private val delegate: akka.persistence.cassandra.session.javadsl.CassandraSession = {
-
-    // FIXME all this can be extracted to core internal and used from both javadsl and scaladsl
-
-    import akka.util.Helpers.Requiring
-    import scala.collection.JavaConverters._
-    import akka.persistence.cassandra.ListenableFutureConverter // implicit asScala conversion
-    val cfg = settings.config
-    val replicationStrategy: String = CassandraPluginConfig.getReplicationStrategy(
-      cfg.getString("replication-strategy"),
-      cfg.getInt("replication-factor"),
-      cfg.getStringList("data-center-replication-factors").asScala
-    )
-
-    val keyspaceAutoCreate: Boolean = cfg.getBoolean("keyspace-autocreate")
-    val keyspace: String = cfg.getString("keyspace").requiring(
-      !keyspaceAutoCreate || _ > "",
-      "'keyspace' configuration must be defined, or use keyspace-autocreate=off"
-    )
-
-    def init(session: Session): Future[Done] = {
-      implicit val ec = executionContext
-      if (keyspaceAutoCreate) {
-        val result1 =
-          session.executeAsync(s"""
-            CREATE KEYSPACE IF NOT EXISTS $keyspace
-            WITH REPLICATION = { 'class' : $replicationStrategy }
-            """).asScala
-        result1.flatMap { _ =>
-          session.executeAsync(s"USE $keyspace;").asScala
-        }.map(_ => Done)
-      } else if (keyspace != "")
-        session.executeAsync(s"USE $keyspace;").asScala.map(_ => Done)
-      else
-        Future.successful(Done)
-    }
+  private implicit val ec = executionContext
 
-    val metricsCategory = "lagom-" + system.name
-    // using the scaladsl API because the init function
-    val scaladslCassandraSession = new akka.persistence.cassandra.session.scaladsl.CassandraSession(
-      system,
-      SessionProvider(system.asInstanceOf[ExtendedActorSystem], settings.config),
-      settings,
-      executionContext,
-      Logging.getLogger(system, this.getClass),
-      metricsCategory,
-      init
-    )
-    new akka.persistence.cassandra.session.javadsl.CassandraSession(scaladslCassandraSession)
-  }
+  /**
+   * Internal API
+   */
+  private[lagom] val delegate: AkkaScalaCassandraSession = CassandraReadSideSessionProvider(system, settings, executionContext)
```

Comment:
Ok, I made a common `CassandraReadSideSessionProvider` to re-use logic around creation so that is why I used the Scala one here, but I'll just have to make factory methods for both of them then.